doc/manual/meson.build

```
@@ -0,0 +1,75 @@
+if get_option('doc')
+  ignore_headers = [
+    'private.h',
+    'pkcs11.h',
+    'conf.h',
+    'debug.h',
+    'dict.h',
+    'filter.h',
+    'log.h',
+    'mock.h',
+    'modules.h',
+    'pkcs11.h',
+    'pkcs11i.h',
+    'pkcs11x.h',
+    'private.h',
+    'client.h',
+    'proxy.h',
+    'rpc.h',
+    'rpc-message.h',
+    'util.h',
+    'virtual.h',
+    'virtual-fixed.h',
+    'array.h',
+    'compat.h'
+  ]
+
+  content_files = [
+    'p11-kit-config.xml', 'p11-kit-sharing.xml',
+    'p11-kit-devel.xml',
+    'p11-kit-proxy.xml',
+    'p11-kit-trust.xml',
+    'p11-kit-remoting.xml',
+    'p11-kit.xml',
+    'pkcs11.conf.xml',
+    'trust.xml',
+    'annotation-glossary.xml'
+  ]
+
+  version_conf = configuration_data()
+  version_conf.set('VERSION', meson.project_version())
+  configure_file(
+    input : 'version.xml.in',
+    output : 'version.xml',
+    configuration : version_conf)
+
+  userdir_conf = configuration_data()
+  userdir_conf.set('p11_user_config', p11_user_config)
+  configure_file(
+    input : 'userdir.xml.in',
+    output : 'userdir.xml',
+    configuration : userdir_conf)
+
+  sysdir_conf = configuration_data()
+  sysdir_conf.set('p11_system_config', p11_system_config)
+  configure_file(
+    input : 'sysdir.xml.in',
+    output : 'sysdir.xml',
+    configuration : sysdir_conf)
+
+  gnome.gtkdoc('p11-kit',
```

When comparing the `autotools` generated documentation and `meson` generated documentation, reference urls are relative instead of absolutes.

This can be fixed by using the `fixxref_args` and setting `--extra-dir` parameter. Setting `--html-dir` is also a good idea.

```suggestion
  glib_prefix = dependency('glib-2.0').get_pkgconfig_variable('prefix')

  fixxref_args = [
    '--html-dir=' + (prefix / gnome.gtkdoc_html_dir(meson.project_name())),
    '--extra-dir=' + (glib_prefix / gnome.gtkdoc_html_dir('glib')),
  ]

  gnome.gtkdoc(
    ...
    fixxref_args: fixxref_args,
    ...
```

---

Thank you, incorporated this.